ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+          register: remaining_rhel7_packages
+          ignore_errors: true
+
+        - name: Remove remaining RHEL7 packages
+          when: not remaining_rhel7_packages.failed
+          dnf:
+            name: "{{ remaining_rhel7_packages.stdout_lines }}"
+            state: absent
+
+        - name: Determine RHEL7 old kernel versions
+          command: "ls -1d *.el7*"
+          args:
+            chdir: /lib/modules
+          register: old_kernel_versions
+          ignore_errors: true
+
+        - name: Remove weak modules from old kernels
+          when: not old_kernel_versions.failed
+          shell: "[ -x /usr/sbin/weak-modules ] && /usr/sbin/weak-modules --remove-kernel {{ item }}"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+        - name: Remove old kernels from bootloader entries
+          when: not old_kernel_versions.failed
+          command: "/bin/kernel-install remove {{ item }} /lib/modules/{{ item }}/vmlinuz"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+    - name: Restore root login only if it was originally enabled
+      when: sshd_config.changed
+      lineinfile:
+        dest: /etc/ssh/sshd_config
+        regexp: '^PermitRootLogin no'
+        line: 'PermitRootLogin yes'
+        state: present
+
+    - *RESTART_SSHD
+
+    - name: Replace epel7 by epel8 repository
+      when:
+        - provider == "non_cloud"
+        - "'Extra Packages for Enterprise Linux 7' in pre_upgrade_active_repositories.stdout"
+      block:
+        - name: Remove epel7 repository
+          package:
+            name: epel-release
+            state: absent
+
+        - name: Install epel8 repository
+          command: "yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm"
+
+    - name: Unset Red Hat subscription manager to consume the latest RHEL content
+      when: provider == 'non_cloud'
+      command: 'subscription-manager release --unset'
+      ignore_errors: true
+
+    - *UPDATE_PACKAGES
+
+    # SELINUX configuration restore
+    - name: Re-enable SELinux
+      when:
+        - not pre_upgrade_selinux_status.stdout == "disabled"
+        - pre_upgrade_selinux_mode == "Enforcing"
+      ansible.posix.selinux:
+        state: enforcing
+
+    - name: Remove cloud-init cache # Initially populated with python2 dependencies that are unmet in RHEL8
+      when: provider == 'azure'
+      block:
+        - name: Find files holding the cloud-init service cache
+          find:
+            paths: /var/lib/cloud/instances
+            patterns: 'obj.pkl'
+            recurse: true
+          register: cloud_init_cache
+
+        - name: Remove cloud-init cache files
+          file:
+            path: "{{ item.path }}"
+            state: absent
+          loop: "{{ cloud_init_cache.files }}"
+
+    - name: Restart cloud-init service # Azure /mnt mount point is cloud-init dependent
+      when: provider == 'azure'
```

Comment:
The same condition is used multiple times sequentially - please use block for grouping.

Reply:
Refactored.